src/defaults/gitignore.txt

```diff
@@ -1,24 +1,426 @@
-# Folders to ignore - files may be too large, too many, etc.
+################################################################################
+#                          DrWatson Project Structure                          #
+################################################################################
+
+# Folders to ignore - files may be too large, too many, etc. NOTE: EDIT IF NEEDED.
 data
 videos
 plots
 notebooks
 _research
 
-# LaTeX build artifacts
-*.pdf
+################################################################################
+#                                    Julia                                     #
+################################################################################
+
+# Files generated by invoking Julia with --code-coverage
+*.jl.cov
+*.jl.*.cov
+
+# Files generated by invoking Julia with --track-allocation
+*.jl.mem
+
+# System-specific files and directories generated by the BinaryProvider and
+# BinDeps packages.
+# They contain absolute paths, and so should not be committed
+deps/deps.jl
+deps/build.log
+deps/downloads/
+deps/usr/
+deps/src/
+
+# Build artifacts for creating documentation generated by the Documenter package
+docs/build/
+docs/site/
+
+# File generated by Pkg, the package manager, based on a corresponding
+# `Project.toml`.  It records a fixed state of all packages used by the project.
+# As such, it should not be committed for packages, but should be committed for
+# applications that require a static environment.
+Manifest.toml
```

Comment:
Manifest.toml is absolutely crucial for reproducibility of scientific results and should always be committed.

Reply:
Will change that, thanks!